data/npc/scripts/bank.lua

```diff
@@ -0,0 +1,339 @@
+		npcHandler:say({
+			"Every citizen has one. The big advantage is that you can access your money in every branch of the World Bank! ...",
+			"Would you like to know more about the {basic} functions of your bank account, the {advanced} functions, or are you already bored, perhaps?"
+		}, cid)
+		npcHandler.topic[cid] = 0
+		return true
+	elseif msgcontains(msg, "balance") then
+		npcHandler.topic[cid] = 0
+		if player:getBankBalance() >= 100000000 then
+			npcHandler:say("I think you must be one of the richest inhabitants in the world! Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		elseif player:getBankBalance() >= 10000000 then
+			npcHandler:say("You have made ten millions and it still grows! Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		elseif player:getBankBalance() >= 1000000 then
+			npcHandler:say("Wow, you have reached the magic number of a million gp!!! Your account balance is " .. player:getBankBalance() .. " gold!", cid)
+			return true
+		elseif player:getBankBalance() >= 100000 then
+			npcHandler:say("You certainly have made a pretty penny. Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		else
+			npcHandler:say("Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		end
+	elseif msgcontains(msg, "deposit") then
+		count[cid] = player:getMoney()
+		if count[cid] < 1 then
+			npcHandler:say("You do not have enough gold.", cid)
+			npcHandler.topic[cid] = 0
+			return false
+		end
+		if msgcontains(msg, "all") then
+			count[cid] = player:getMoney()
+			npcHandler:say("Would you really like to deposit " .. count[cid] .. " gold?", cid)
+			npcHandler.topic[cid] = 2
+			return true
+		else
+			if string.match(msg,"%d+") then
+				count[cid] = getMoneyCount(msg)
+				if count[cid] < 1 then
+					npcHandler:say("You do not have enough gold.", cid)
+					npcHandler.topic[cid] = 0
+					return false
+				end
+				npcHandler:say("Would you really like to deposit " .. count[cid] .. " gold?", cid)
+				npcHandler.topic[cid] = 2
+				return true
+			else
+				npcHandler:say("Please tell me how much gold it is you would like to deposit.", cid)
+				npcHandler.topic[cid] = 1
+				return true
+			end
+		end
+		if not isValidMoney(count[cid]) then
+			npcHandler:say("Sorry, but you can't deposit that much.", cid)
+			npcHandler.topic[cid] = 0
+			return false
+		end
+	elseif npcHandler.topic[cid] == 1 then
+		count[cid] = getMoneyCount(msg)
+		if isValidMoney(count[cid]) then
+			npcHandler:say("Would you really like to deposit " .. count[cid] .. " gold?", cid)
+			npcHandler.topic[cid] = 2
+			return true
+		else
+			npcHandler:say("You do not have enough gold.", cid)
+			npcHandler.topic[cid] = 0
+			return true
+		end
+	elseif npcHandler.topic[cid] == 2 then
+		if msgcontains(msg, "yes") then
+			if player:getMoney() >= tonumber(count[cid]) then
+				player:depositMoney(count[cid])
```

Comment:
I'm not sure, but wouldn't it be nice to place a conditional check here?

Reply:
you mean, taking it off of the depositMoney function?:

https://github.com/otland/forgottenserver/pull/2934/files#diff-b6b032d21c948d15f9e768a6481cdd98R135-R142

Reply:
no, just check it out: if not function then

Reply:
sorry, I still don't understand 😢 
can you re-do this for me here?
```
	elseif npcHandler.topic[cid] == 2 then
		if msgcontains(msg, "yes") then
			if player:getMoney() >= tonumber(count[cid]) then
				player:depositMoney(count[cid])
				npcHandler:say("Alright, we have added the amount of " .. count[cid] .. " gold to your {balance}. You can {withdraw} your money anytime you want to.", cid)
			else
				npcHandler:say("You do not have enough gold.", cid)
			end
		elseif msgcontains(msg, "no") then
			npcHandler:say("As you wish. Is there something else I can do for you?", cid)
		end
		npcHandler.topic[cid] = 0
		return true
```